### PR TITLE
fix: re2 dot all flag for non-hogql filters

### PR DESCRIPTION
**Pull request by pauldambra**
we follow re2 regex syntax and so does ClickHouse **except**
For example, the string a\nb shouldn't match the pattern a.b, but it does in CH
this is because According to the re2 docs, the s flag is false by default,
but in CH it seems to be true by default.
prepending (?-s) to the regex string will make it work as expected
see https://github.com/ClickHouse/ClickHouse/issues/34603

----

we found this while investigating queries hanging here https://posthog.slack.com/archives/C03PB072FMJ/p1703123107173919

We don't completely understand why this fixes it but in the reproduction case we had the query would hang forever (eventually killing the node) but not when prepending `(?-s)` to the query 

I _think_ this patches regex for both non-HogQL property filters and HogQL queries

----

in our docs we explicitly say dot won't match newlines https://posthog.com/tutorials/regex-basics#the-basics-of-regex so it should be safe (or at least more correct) to change this behaviour for all filters

### CODE REVIEW

**Comment by fuziontech** [comment]
this looks right....let me tag an expert though

File: posthog/hogql/property.py [hunk 1]
```diff
@@ -202,12 +202,18 @@ def property_to_expr(
...
             )
+        # we follow re2 regex syntax and so does ClickHouse **except**
+        # For example, the string a\nb shouldn't match the pattern a.b, but it does in CH
+        # this is because According to the re2 docs, the s flag is false by default,
+        # but in CH it seems to be true by default.
+        # prepending (?-s) to the regex string will make it work as expected
+        # see https://github.com/ClickHouse/ClickHouse/issues/34603
         elif operator == PropertyOperator.regex:
-            return ast.Call(name="match", args=[field, ast.Constant(value=value)])
+            return ast.Call(name="match", args=[field, ast.Constant(value=f"(?-s){value}")])
```
Comment (fuziontech) [inline posthog/hogql/property.py:212]: This definitely looks correct.

File: posthog/models/property/util.py [hunk 2]
```diff
@@ -489,7 +489,13 @@ def prop_filter_json_extract(
...
         params = {
             "k{}_{}".format(prepend, idx): prop.key,
-            "v{}_{}".format(prepend, idx): prop.value,
+            # we follow re2 regex syntax and so does ClickHouse **except**
+            # For example, the string a\nb shouldn't match the pattern a.b, but it does in CH
+            # this is because According to the re2 docs, the s flag is false by default,
+            # but in CH it seems to be true by default.
+            # prepending (?-s) to the regex string will make it work as expected
+            # see https://github.com/ClickHouse/ClickHouse/issues/34603
+            "v{}_{}".format(prepend, idx): f"(?-s){prop.value}",
```
Comment (fuziontech) [inline posthog/models/property/util.py:498]: pretty sure this is right

**Review by fuziontech — APPROVED** [review]

**Comment by pauldambra** [comment]
@mariusandra for the HogQL assist too?

**Review by EDsCODE — APPROVED** [review]
Definitely not a regex expert but based on the issue this looks right. Maybe worth having a test

**Comment by fuziontech** [comment]
Closing this one up since we won't be landing it (it breaks regex elsewhere in the app)

**Pull request closed without merging**